pkgs/applications/window-managers/dwl/default.nix

```diff
@@ -1,36 +1,44 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, installShellFiles
+, libX11
 , libinput
 , libxcb
 , libxkbcommon
 , pixman
 , pkg-config
+, substituteAll
 , wayland
 , wayland-protocols
-, wlroots
+, wlroots_0_16
 , writeText
-, enable-xwayland ? true, xwayland, libX11
+, xcbutilwm
+, xwayland
+, enableXWayland ? true
 , conf ? null
 , patches ? [ ]
 }:
 
 let
   totalPatches = patches ++ [ ];
+  wlroots = wlroots_0_16;
```

Comment:
This could probably go in `callPackage`.

Reply:
Reference: https://github.com/NixOS/nixpkgs/issues/204303